File: Data Structures & Algorithms/climbing-stairs/submission-2.py
def dp(n: int) -> int:
    if n < 0:
        return 0
    if n == 0:
        return 1
    return dp(n-2) + dp(n-1)

class Solution:
    def climbStairs(self, n: int) -> int:
        return dp(n)

File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let seen = {}
        for (let i=0; i < nums.length; i++) {
            let num = nums[i]
            let other_num = target - num
            if (other_num in seen) {
                return [seen[other_num], i]
            }
            seen[num] = i
        }
    }
}
